src/main/scala/wdlTools/syntax/Antlr4Util.scala

```diff
@@ -111,13 +114,111 @@ object Antlr4Util {
     }
   }
 
-  case class Grammar[L <: Lexer, P <: Parser](lexer: L,
-                                              parser: P,
-                                              errListener: WdlAggregatingErrorListener,
-                                              comments: mutable.Map[Int, Comment],
-                                              docSourceUrl: Option[URL] = None,
-                                              opts: Options) {
-    def verify(): Unit = {
+  trait ParseTreeListenerFactory {
+    def createParseTreeListeners(grammar: Grammar): Vector[ParseTreeListener]
+  }
+
+  private case class CommentListenerFactory() extends ParseTreeListenerFactory {
```

Comment:
I don't know if you want to, but you can use `private case object CommentListenerFactory extends ParseTreeListenerFactory` instead. In other words, replace the class with an object.

Reply:
thanks, That's better